File: protocol/src/test/typescript/zfoots/packet/EmptyObject.ts
import IByteBuffer from '../IByteBuffer';
import IProtocolRegistration from '../IProtocolRegistration';


class EmptyObject {
    
}

export class EmptyObjectRegistration implements IProtocolRegistration<EmptyObject> {
    protocolId(): number {
        return 0;
    }

    write(buffer: IByteBuffer, packet: EmptyObject | null) {
        if (packet === null) {
            buffer.writeInt(0);
            return;
        }
        buffer.writeInt(-1);
    }

    read(buffer: IByteBuffer): EmptyObject | null {
        const length = buffer.readInt();
        if (length === 0) {
            return null;
        }
        const beforeReadIndex = buffer.getReadOffset();
        const packet = new EmptyObject();
        
        if (length > 0) {
            buffer.setReadOffset(beforeReadIndex + length);
        }
        return packet;
    }
}

export default EmptyObject;

File: protocol/src/test/dart/zfoodart/packet/EmptyObject.dart
import '../IProtocolRegistration.dart';
import '../IByteBuffer.dart';

class EmptyObject {
  
}

class EmptyObjectRegistration implements IProtocolRegistration<EmptyObject> {
  @override
  int protocolId() {
    return 0;
  }

  @override
  void write(IByteBuffer buffer, EmptyObject? packet) {
    if (packet == null) {
      buffer.writeInt(0);
      return;
    }
    buffer.writeInt(-1);
  }


  @override
  EmptyObject read(IByteBuffer buffer) {
    var length = buffer.readInt();
    var packet = EmptyObject();
    if (length == 0) {
      return packet;
    }
    var beforeReadIndex = buffer.getReadOffset();
    
    if (length > 0) {
      buffer.setReadOffset(beforeReadIndex + length);
    }
    return packet;
  }
}